### PR TITLE
Draw attention to 3rd step in process to update so numbers for release.

### PR DESCRIPTION
**Pull request by lrknox**
resetting LT_VERS_AGE was missed for the last release.   The 3rd step instruction is now clearly separated from the second step instruction and the steps are numbered.  A step describing the update is also added to the RELEASE_PROCESS.md file.

### CODE REVIEW

**Review by byrnHDF — APPROVED** [review]

File: release_docs/RELEASE_PROCESS.md [hunk 1]
```diff
@@ -54,10 +54,11 @@ For more information on the HDF5 versioning and backward and forward compatibili
...
+3. Be sure to complete all four steps to update so numbers for each deployed lib file in the process described in config/lt_vers.am and check that the .so numbers for lib files in binaries correctly indicate compatibility status with the previous release.  
+4. Move all unresolved Milestone issues to the next release version in GitHub.
+5. Verify that frozen code branch satisfies all existing regression test cases, and give the 'OK' to the release coordinator once all daily test configurations are passing as expected after the date of the code freeze. If there are failing tests after the code freeze date, coordinate with maintainers responsible for the failures to ensure that either the changes causing the failures are corrected or reverted. 
+6. Verify release branches for third-party software used: SZIP, ZLIB, and Plugins; and announce release versions to hdf5lib@lists.hdfgroup.org.
```
Comment (hyoklee) [inline release_docs/RELEASE_PROCESS.md]: hdf5lib@hdfgroup.org

**Review by hyoklee — CHANGES_REQUESTED** [review]
Drop `lists` from email address.

File: release_docs/RELEASE_PROCESS.md (hunk 1 above)
Comment (lrknox) [inline release_docs/RELEASE_PROCESS.md:61]: 6. Verify release branches for third-party software used: SZIP, ZLIB, and Plugins; and announce release versions to hdf5lib@lhdfgroup.org.
```suggestion
6. Verify release branches for third-party software used: SZIP, ZLIB, and Plugins; and announce release versions to hdf5lib@lists.hdfgroup.org.
```

Comment (lrknox) [inline release_docs/RELEASE_PROCESS.md]:
```suggestion
6. Verify release branches for third-party software used: SZIP, ZLIB, and Plugins; and announce release versions to hdf5lib@hdfgroup.org.
```

**Review by mattjala — APPROVED** [review]

**Review by hyoklee — APPROVED** [review]

**Pull request merged**